File: Internal/Common/Channel.cs
using System.Text.Json.Serialization;
using BonfireServer.Database;

namespace BonfireServer.Internal.Common;

public class Channel(LiteFlakeId? id) : ICachableType
{
    public LiteFlakeId Id { get; set; } =  id ?? new LiteFlakeId();

    public string Name { get; set; }

    public Server? Server { get; set; } = null;
    [JsonIgnore] public List<Message> Messages { get; set; } = [];
}

File: Public/Scripts/Common/Channel.ts
import {Message} from "./Message";
import {Server} from "./Server";

export class Channel {
    public Id: number;
    
    public Name: string;
    public Server: Server | null;
    public Messages!: Map<number, Message>;
    
    constructor(
        id: number,
        name: string,
        server: Server | null, 
        messages: Message[]
    ) {
        this.Id = id;
        this.Name = name;
        this.Server = server;
        this.Messages = new Map();
        
        messages.map(x => this.Messages.set(x.Id, x));
    }
}